### ISSUE TITLE
Problem with movie muxing and burning oin disc

### ISSUE THREAD

**Issue by wolflopnew** [justdan96/tsMuxer#596]
Movie does not start on player after creation with the Tsmuxer or only after 1 min. Have upscale mp4 and AAC sound loaded. After adding in the muxer and adding additional audio tracks (DTS) and clicking on muxing it shows 'Warning! AAC codec is not standard for BD disks, the m2ts will not play on a Blu-ray player.' But the film is still muxxed. If I omit the AAC soundtrack when muxing due to the warning, the parse error 'Not enough buffer for parse video stream' comes up again. Current frame number 78'.
So I muxx with AAC. Film can be played on PC. If you load this created iso again in muxer and then omit the AAC, the iso will be created without any problems. However, the film cannot be played after burning the iso. The message play then stop appears. If I load this film in clonebd/DVDFab and shrink it by a few MBs, then the film is playable.
Many more bugs. Please help!

**Comment by jcdr428** [comment]
@wolflopnew please demux the hevc stream from the mp4 (eg with "`ffmpeg.exe -i input.mp4 -c:v copy output.hevc`") and upload the .hevc.
No ISO or ZIP please: I will download the first 80 frames only, not the full UHD.

**Comment by wolflopnew** [comment]
Still another problem i tried to solve the problem by myself is load the hevc in scenerist. But here comes the following error message. https://1drv.ms/u/s!AifiITtpdjN1h32fTGFsHHUOcIcb?e=59D1dg
Scenerist allows only uhd compatible files So the build file is not uhd coimpatible. Can you also solve this problem? So if the hevs is loadable to scenerist. Movie is always playable and no problems with starting after 1 min.

**Comment by wolflopnew** [comment]
here is the link to the hevc 3 Min. from teh film
https://1drv.ms/u/s!AifiITtpdjN1h36Kkir65vyO4CMB?e=1CKH6C

**Comment by jcdr428** [comment]
@wolflopnew nothing wrong with the hevc, it looks like I will have to download the whole mp4. Can you please share a link.

**Comment by wolflopnew** [comment]
I have the whole file anymore. i will send  you an other movie file with the same problem.  But the mp4 are maybe 45 Gb and more  in size. Dont know how to send. They are upscales 4k.

**Comment by wolflopnew** [comment]
Here is anonther link with a 3 Min Movie same problems as i described
https://1drv.ms/u/s!AifiITtpdjN1h3-R3_gSzsIS20g4?e=NoLdUs

**Comment by wolflopnew** [comment]
for comparison the executable version generated by clonebd from the (TSmuxer version sample above)
Possibly for comparison. Maybe that gives a clue as to why...
https://1drv.ms/u/s!AifiITtpdjN1iACtSV8YQgEw0TiW?e=iu5txE

**Comment by jcdr428** [comment]
@wolflopnew the provided sample does not result in any warning in tsMuxer when muxing on my PC, so nothing I can do.

**Comment by wolflopnew** [comment]
But if they were burned on a disc they are not playable on the player?

**Comment by wolflopnew** [comment]
by the way have ypu tried to mux the video without the aac? Muxer stops with an error.

**Comment by wolflopnew** [comment]
heres todays actual encoding... erros message
https://1drv.ms/u/s!AifiITtpdjN1iAE91XBfumQDPhX7?e=7wLMZr

**Comment by wolflopnew** [comment]
without aac this error message
https://1drv.ms/u/s!AifiITtpdjN1iALmxEroSTJE1Mls?e=J2eH2L

**Comment by wolflopnew** [comment]
So the same problem today Disc is not playable after burning it. Player showy only play at display and stops then... 
So if to solve the problem you must look burned disc on a stand alone player. Did anyone can test ist on a burner an player to confirm my problem?
Here is the today tested file
https://1drv.ms/u/s!AifiITtpdjN1iAMDiLcK5NaWLV9i?e=e6vUzY

**Comment by jcdr428** [comment]
> by the way have ypu tried to mux the video without the aac? Muxer stops with an error.

The sample you provided does not have any AAC stream.
Please upload a faulty sample. Again, no point uploading working samples.

**Comment by wolflopnew** [comment]
Here ist the mp4 Upcsale shortintro -  with all the problems i described... Hope this now the right one
https://1drv.ms/v/s!AifiITtpdjN1iAWgDRSvnTuRG6Nn?e=pwjRuT

**Comment by jcdr428** [comment]
@wolflopnew please try https://github.com/jcdr428/tsMuxer/suites/6627064564/artifacts/249768855 (assuming you're on Win64).
Put the downloaded tsMuxeR.exe file with your tsMuxerGUI.exe file.

**Comment by wolflopnew** [comment]
Thanks, i make a test with it

**Comment by wolflopnew** [comment]
problem with AAC Audio is gone. Film is muxxable but Still the problem... movie is, after burning the iso, not playable on the standalone player

**Comment by jcdr428** [comment]
@wolflopnew Blu-ray is not compatible with aac, nothing we can do...
I'll push the commit to fix the ''Not enough buffer for parse video stream' error.

**Comment by wolflopnew** [comment]
have burned the movie with DVD Fab  but this encodes without aac automatically . So the movie is smaler but it playes normal on the player.  I tried antother run with hooked aac. I will burn this an i give you feedback...

**Comment by wolflopnew** [comment]
So my feedback... 4k Blu-ray is after encoding the movie with aac and dts audio playable.  The aac has no sound. The player takes automatic the dts sound audio and it worke fine... So why not without antohter encoding?

**Comment by jcdr428** [comment]
@wolfopnew can you please share a (small) sample of the m2ts with aac that works in the player ? I will analyse the aac descriptor written by DVDFab.

**Comment by wolflopnew** [comment]
Here is the working movielink https://1drv.ms/u/s!AifiITtpdjN1iAbbdFlcSyVTvdsR?e=FPGPpa

**Comment by jcdr428** [comment]
@wolflopnew the m2ts does not contain any AAC stream, only AC3. So it is likely that DVDFab transcodes AAC to AC3.
tsMuxer is a muxer not a transcoder. You can use a transcoding tool e.g. foobar2000 or ffmpeg to transcode AAC before muxing with tsMuxer.

**Comment by wolflopnew** [comment]
But i dont need aac audio its only ballast. I need only the second audio DTS .- Master. So i dont need a trranscoding. I need the hevc and the DTS MA, be muxed burnable on disc and playable with the standalone player and this doesnt work.

**Comment by jcdr428** [comment]
@wolflopnew there are two AC3 on the sample you provided, no DTS. What do you wnat me to verify with it ?

**Comment by wolflopnew** [comment]
ok,  with this movie there are two ac3. Not seen, sorry. But the first was the aac ( dvdfab trancodes it ti ac3) but the second ist the original track i take at the end.

**Comment by jcdr428** [comment]
@wolflopnew still not clear: 
DVDFab transcodes AAC to AC3, and tsMuxer will not do that.
If there is a stream with DTS that does not work with tsMuxer but works with DVDFab, please provide both tsMuxer and DVDFab outputs so that I can analyse the reason.

**Comment by wolflopnew** [comment]
But does this solve the problem? AAC is not the problem.  I explained to you earlier.... You already have all of the required files for troubleshooting... 
I have an mp4 that has been upscaled to 4k which then contains an Aac audio. I load this mp4 into the muxer and disable the aac (because it's not needed) and add a high quality DTMMA audio track and the subtitles and the original chapters. The generated Iso/BdmV should be burned and played on the standalone player. If I burn the e.g. iso generated by muxer, it cannot be played. If I load this iso into clonebd and compress it minimally (only copies the 1:1), then the burned iso created with clone can be played.

**Comment by jcdr428** [comment]
@wolflopnew as requested, please upload the tsMuxer iso that does not play and the DVDFab iso (from the same input) that plays. Up to now if I am not mistaken you have not provided them.

**Comment by wolflopnew** [comment]
I will send tomorrow another projectfile as you asked. Older files are deleted...

**Comment by wolflopnew** [comment]
Here the isos from the current project, 
https://1drv.ms/u/s!AifiITtpdjN1iAhyAMHItAfBNozf?e=hzNOQf
the Hitcut.ISO one is created with the muxer from an Mp4 with aac and can be played on the player.
Film starts but only after 1 minute.

Hitiso2 is the Mp4 without AAC checked and with DTS sound and sups but only 2 MB. Tsmuxer ends without an error... but 2 MBs?
https://1drv.ms/u/s!AifiITtpdjN1iAebDPWW9bDCNonl?e=iQTuZl
Hitiso2a tried a second time... take the mlp form the whole muxer has mad and her it gives me normal size. Its the cut from the original Disc i expect  from the muxer to play normal. But it starts at 1 Min. but also with  Dvdfab it starts at 1 Min...?
https://1drv.ms/u/s!AifiITtpdjN1iArC4h8PQFLbogGa?e=iBhDdo


hitcut 3 folder.rar is created with DVDfab from Hitcut.Iso and DVDfab UhdCreator. The film runs smoothly on the player from 0.00 min. but ypu see its smaler while compressed (transcoded).
https://1drv.ms/u/s!AifiITtpdjN1iAnTPzahpJEU6dJ4?e=IWOUhp

I hope the real-time help in finding the error, thank you very much

**Comment by jcdr428** [comment]
@wolflopnew can you please also share the original mp4, the DTS and the sups used as input to create those outputs, so that I can reproduce.

**Comment by wolflopnew** [comment]
I have made a sequence of 2,5 GB from the original Mp4. So give me an info if you had downloaded the first files so that i can upload the mp4 file and then the dts file

**Comment by wolflopnew** [comment]
The Dts file upload if you ready with the other files

**Comment by wolflopnew** [comment]
Today had begin another movie. And as before player shows play... nothing to sees on the tv. After a while movie starts at 1 Minute.
So i think the key is  why it starts after 1 Minute and when the movie is compressed for a few bytes with clonebd it will start normal...

**Comment by jcdr428** [comment]
@wolflopnew I can't find the reason for the 1min. offset. Except maybe the unusually large initial time offset used by tsMuxer ?
Which I have reduced to 10 sec., please try https://github.com/justdan96/tsMuxer/suites/6701592582/artifacts/254746586

I have also corrected German language code from "ger" to "deu", however I doubt this was the reason for your player behavior.

**Comment by wolflopnew** [comment]
Just tested that. No change. The film starts on February 1st. Min. It's funny that this only happens with all Upscale Mp4 with movies created with the Tsmuxer. For regular hevc movies from disc, creating isos works normally.

**Comment by wolflopnew** [comment]
Did you still need the DTS audio file?

**Comment by jcdr428** [comment]
@wolflopnew I have the dts from Hitcut3, that should do.
I've also downloaded the mp4, i'll keep on digging.

**Comment by wolflopnew** [comment]
thanks for your help. i will delete now the files on one drive...

**Comment by jcdr428** [comment]
@wolflopnew can you please see whether replacing the index.bdmv and MovieObject.bdmv in the tsMuxer output, by the ones produced by DVDFab, resolves the issue ? This would allows me to know for sure that it comes from the BDMV commands.

**Comment by wolflopnew** [comment]
still the same problem even if I copy the index.bdmv and MovieObject.bdmv from the DVdfab. Film starts at 1 min. The dVdfab starts at 0 min.

**Comment by wolflopnew** [comment]
> Movie does not start on player after creation with the Tsmuxer or only after 1 min. Have upscale mp4 and AAC sound loaded. After adding in the muxer and adding additional audio tracks (DTS) and clicking on muxing it shows 'Warning! AAC codec is not standard for BD disks, the m2ts will not play on a Blu-ray player.' But the film is still muxxed. If I omit the AAC soundtrack when muxing due to the warning, the parse error 'Not enough buffer for parse video stream Current frame number 78' comes up again. So I muxx with AAC. Film can be played on PC. If you load this created iso again in muxer and then omit the AAC, the iso will be created without any problems. However, the film cannot be played after burning the iso with Imgburn. The message play then stop appears. If I load this film in clonebd/DVDFab and shrink it by a few MBs, then the film is playable. Many more bugs. Please help!

i think i have the solution. It was due to the frame rate Adobe Media Encoder generated 23.9761
frames instead of 23.976. Hence the problems...
Is it possible to change the frames to 23.976. However, the 23.9761 is displayed in the drop-down menu.

![Screenshot 2022-11-02 102250](https://user-images.githubusercontent.com/101669182/199453386-516fcb57-56e7-43c5-a3ea-c26a04bc7f16.jpg)
